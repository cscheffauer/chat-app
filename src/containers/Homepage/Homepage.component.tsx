import React, { useState } from 'react'
import LogIn from '../../components/LogIn/LogIn.component';
import ChatContainer from '../ChatContainer/ChatContainer.component';
import Header from '../../components/Header/Header.component';

import './Homepage.scss';

type Props = {
    client: any
}


const Homepage = ({ client }: Props) => {
    const [loggedIn, setloggedIn] = useState(false)
    const [username, setusername] = useState("" as String);

    const logInUser = (username: String) => {
        setusername(username);
        client.send(JSON.stringify({
            username,
            type: "userevent"
        }));

        setloggedIn(true);
    };


    return (
        <div className="homepage">
            {
                !loggedIn ?
                    <LogIn logInUser={logInUser} />
                    :
                    <ChatContainer client={client} username={username} />
            }
        </div>
    )
}

export default Homepage
