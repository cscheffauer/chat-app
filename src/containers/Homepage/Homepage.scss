@import '../../variables';

.homepage {
	width: 100%;
	height: 92vh; /* Fallback for browsers that do not support Custom Properties */
	height: calc(var(--vh, 1vh) * 92);
	@media only screen and (min-width: $chatwindowWidth) {
		width: $chatwindowWidth;
		position: absolute;
		left: calc(50% - #{$chatwindowWidth} / 2);
	}
}
