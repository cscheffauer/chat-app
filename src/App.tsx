import React, { useState } from 'react';
import { w3cwebsocket as W3CWebSocket } from "websocket";

import './index.scss';
import Homepage from './containers/Homepage/Homepage.component';
import Header from './components/Header/Header.component';


export const client = new W3CWebSocket('ws://localhost:8001');


const App = () => {
  const [connection, setconnection] = useState('loading');

  client.onopen = () => {
    setconnection('established');
    console.log('WebSocket Client Connected');
  };

  client.onerror = () => {
    setconnection('error');
    console.log("Can't connect to Websocket Server.")
  }

  const renderConnectionSwitch = () => {
    switch (connection) {
      case 'loading':
        return <p>Loading...</p>
      case 'established':
        return <Homepage client={client} />
      case 'error':
        return <div className="errorboundary">< h3 >Chat is currently unavailable!</h3 > <p>Pls contact support.</p></div >
      default:
        return <div />
    }
  }

  return (
    <>
      <Header />
      {
        renderConnectionSwitch()
      }
    </>
  );

}

export default App;