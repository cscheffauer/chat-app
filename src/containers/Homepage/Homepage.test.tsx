import React from 'react';
import { shallow } from 'enzyme';
import Homepage from './Homepage.component';

import { client } from '../../App';

describe('Container / Homepage', () => {
    const clientMock = client;
    const HomepageWrapper = shallow(<Homepage client={clientMock} />);

    it('expect to render Homepage container', () => {
        expect(HomepageWrapper).toMatchSnapshot();
    })
})


