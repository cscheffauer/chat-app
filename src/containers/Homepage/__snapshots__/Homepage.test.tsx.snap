// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container / Homepage expect to render Homepage container 1`] = `
<div>
  <LogIn
    logInUser={[Function]}
  />
</div>
`;
